src/libasr/codegen/wasm_to_x64.cpp

```diff
@@ -40,19 +41,105 @@ class X64Visitor : public WASMDecoder<X64Visitor>,
         wasm_bytes.from_pointer_n(code.data(), code.size());
     }
 
+    void visit_Return() {}
+
+    void call_imported_function(uint32_t func_idx) {
+        switch (func_idx) {
+            case 0: {  // print_i32
+                std::cerr << "Call to print_i32() will be printed as exit code\n";
+
+                // Currently, for print we are setting the value to be printed
+                // as the exit code. Later we can access/view this value from console
+                // using: echo $?
+                m_a.asm_pop_r64(X64Reg::rdi); // get exit code from stack top
+                m_a.asm_mov_r64_imm64(X64Reg::rax, 60); // sys_exit
+                m_a.asm_syscall(); // syscall
+                break;
+            }
+            case 1: {  // print_i64
+                std::cerr << "Call to print_i64() is not yet supported\n";
+                break;
+            }
+            case 2: {  // print_f32
+                std::cerr << "Call to print_f32() is not yet supported\n";
+                break;
+            }
+            case 3: {  // print_f64
+                std::cerr << "Call to print_f64() is not yet supported\n";
+                break;
+            }
+            case 4: {  // print_str
+                std::cerr << "Call to print_str() is not yet supported\n";
+                break;
+            }
+            case 5: {  // flush_buf
+                std::cerr << "Call to flush_buf() is not yet supported\n";
+                break;
+            }
+            case 6: {  // set_exit_code
+                m_a.asm_pop_r64(X64Reg::rdi); // get exit code from stack top
+                m_a.asm_mov_r64_imm64(X64Reg::rax, 60); // sys_exit
+                m_a.asm_syscall(); // syscall
+                break;
+            }
+            default: {
+                std::cerr << "Unsupported func_idx";
+            }
+        }
+    }
+
+    void visit_Call(uint32_t func_idx) {
+        if (func_idx <= 6U) {
+            call_imported_function(func_idx);
+            return;
+        }
+
+        func_idx -= 7u; // adjust function index as per imports
+        m_a.asm_call_label(exports[func_idx].name);
+    }
+
+    void visit_I32Const(int32_t value) {
+        // direct addition of imm64 to stack is not available with us yet
+        // so temporarily using a combination of instructions
+        // TODO: Update this once we have support for push_imm64()
+        m_a.asm_mov_r64_imm64(X64Reg::rax, value);
+        m_a.asm_push_r64(X64Reg::rax);
+    }
```

Comment:
**doubt:** We are building a `64-bit` assembly backend. Therefore, for `visit_I32Const()`, should we be adding a `32-bit` constant or a `64-bit` constant to the stack? At the moment, I used a `64-bit` constant as in the above code.

Reply:
As long as we correctly implement the WASM semantics for all types (in this case I32), then we can do whatever we want. 

I think the 64bit x86 supports using 32-bit values, so we should use it I think.

Reply:
Got it. Thank you.